Fixed #30934 - Include database alias in `django.db.backends` debug output

codeinthehole opened the pull request:
This is useful if you're working with database routers and want to inspect 
where SQL statements are being routed to.

codeinthehole commented on django/db/backends/utils.py:
```diff
@@ -123,10 +123,11 @@ def debug_sql(self, sql=None, params=None, use_last_executed_query=False, many=F
                 sql,
                 params,
+                self.db.alias,
                 extra={'duration': duration, 'sql': sql, 'params': params},
```
Good point. Will address your comments.

pope1ni commented on tests/backends/test_utils.py:
```diff
@@ -90,3 +91,25 @@ def test_unsupported_callproc_kparams_raises_error(self):
+
+
+class CursorDebugWrapperTests(SimpleTestCase):
+    @mock.patch("django.db.backends.utils.logger")
```
Please use single quotes throughout.

codeinthehole replied:
I'm surprised there isn't a linting test for this kind of thing. Seems a waste of brain cycles for code reviewers to have to look for stylistic errors.

pope1ni commented on tests/backends/test_utils.py:
```diff
@@ -90,3 +91,25 @@ def test_unsupported_callproc_kparams_raises_error(self):
+    def test_debug_sql_logging(self, logger):
+        cursor = mock.Mock()
+
+        # Create a mock database instance including the wrap_database_errors context manager
```
Add a period to the end of all comments.

pope1ni commented on tests/backends/test_utils.py:
```diff
@@ -90,3 +91,25 @@ def test_unsupported_callproc_kparams_raises_error(self):
+        db = mock.MagicMock(alias="mydb", execute_wrappers=[])
+        db.wrap_database_errors.__exit__.return_value = True
+
+        # Execute a dummy query using the wrapper class
```
Drop this comment.

pope1ni commented on tests/backends/test_utils.py:
```diff
@@ -90,3 +91,25 @@ def test_unsupported_callproc_kparams_raises_error(self):
+        wrapper = CursorDebugWrapper(cursor, db)
+        wrapper.execute(sql="SELECT 1;")
+
+        # Check the logged messages include the alias
```
Drop this comment.

pope1ni commented:
@codeinthehole Oops! Can you update the PR title and commit message to `Fixed #30934 -- ...` instead of `Fixed #30910 -- ...`? Thanks.

codeinthehole commented:
@pope1ni Have addressed your comments (good spot on the ticket number) and squashed into the original commit. Let me know if there's anything else required.

pope1ni commented:
@codeinthehole: Could you also fix the ticket number in the PR title?

Thanks, it looks good to me. The ticket is on the [review queue](https://code.djangoproject.com/query?status=!closed&needs_better_patch=0&needs_tests=0&needs_docs=0&has_patch=1&stage=Accepted&desc=1&order=changetime) - one of the fellows will pick it up when they get to it.

felixxm commented on tests/backends/test_utils.py:
```diff
@@ -90,3 +91,23 @@ def test_unsupported_callproc_kparams_raises_error(self):
+        self.assertEqual(db.alias, call_args[4])
+
+        extra = logger.debug.call_args[1]['extra']
+        self.assertEqual(db.alias, extra['alias'])
```
IMO this tests should be moved to the `tests.backends.tests.BackendTestCase`, also we don't need to mock a database, checking for a `connection.alias` in a logger should be enough, e.g.
```diff
diff --git a/tests/backends/tests.py b/tests/backends/tests.py
index da20d94442..5f6e02d91d 100644
--- a/tests/backends/tests.py
+++ b/tests/backends/tests.py
@@ -3,6 +3,7 @@ import datetime
 import threading
 import unittest
 import warnings
+from unittest import mock
 
 from django.core.management.color import no_style
 from django.db import (
@@ -492,6 +493,23 @@ class BackendTestCase(TransactionTestCase):
             BaseDatabaseWrapper.queries_limit = old_queries_limit
             new_connection.close()
 
+    @mock.patch('django.db.backends.utils.logger')
+    @override_settings(DEBUG=True)
+    def test_queries_logger(self, mocked_logger):
+        sql = 'SELECT 1' + connection.features.bare_select_suffix
+        with connection.cursor() as cursor:
+            cursor.execute(sql)
+        params, kwargs = mocked_logger.debug.call_args
+        self.assertIn('; alias=%s', params[0])
+        self.assertEqual(params[2], sql)
+        self.assertEqual(params[3], None)
+        self.assertEqual(params[4], connection.alias)
+        self.assertEqual(
+            list(kwargs['extra']),
+            ['duration', 'sql', 'params', 'alias'],
+        )
+        self.assertEqual(tuple(kwargs['extra'].values()), params[1:])
+
     def test_timezone_none_use_tz_false(self):
         connection.ensure_connection()
         with self.settings(TIME_ZONE=None, USE_TZ=False):
```

jacobtylerwalls commented on docs/topics/logging.txt:
```diff
@@ -568,6 +568,9 @@ Messages to this logger have the following extra context:
 * ``params``: The parameters that were used in the SQL call.
+* ``alias``: The alias of the database used in the SQL call.
+  
+  .. versionchanged:: 3.2
```
```suggestion
  .. versionchanged:: 4.0
```

jacobtylerwalls commented on tests/backends/test_utils.py:
```diff
@@ -3,7 +3,7 @@
 from django.db import NotSupportedError, connection
 from django.db.backends.utils import (
-    format_number, split_identifier, truncate_name,
+    DebugWrapper, format_number, split_identifier, truncate_name,
```
```suggestion
    format_number, split_identifier, truncate_name,
```

jacobtylerwalls reviewed (commented):
Hi @codeinthehole, thanks for this. It looks like this languished due to some "waiting for author" flags on the ticket. One flag is set because we still need a one-sentence release note in `docs/releases/4.0.txt` (under miscellaneous?), but the other I think you've already addressed.

If you're down for adding the release note and then unchecking all "needs" flags on the ticket I imagine we could get this in quickly. Much appreciated.

felixxm commented:
Closing due to inactivity.
